src/urllib3/response.py

```diff
@@ -151,7 +160,184 @@ def _get_decoder(mode):
     return DeflateDecoder()
 
 
-class HTTPResponse(io.IOBase):
+class BaseHTTPResponse(io.IOBase):
+    CONTENT_DECODERS = ["gzip", "deflate"]
+    if brotli is not None:
+        CONTENT_DECODERS += ["br"]
+    REDIRECT_STATUSES = [301, 302, 303, 307, 308]
+
+    DECODER_ERROR_CLASSES = (IOError, zlib.error)
+    if brotli is not None:
+        DECODER_ERROR_CLASSES += (brotli.error,)
+
+    def __init__(
+        self,
+        headers=None,
+        status=0,
+        version=0,
+        reason="",
+        decode_content=True,
+        enforce_content_length=False,
```

Comment:
Should we avoid mentioning the default values here? It sounds likely we'll update one class and not the other one in the future.

Also, should those be kwargs only?

Reply:
Oh yeah, we can do that now can't we :) I'll remove the defaults and add types+keyword-only.